### PR TITLE
Upgrade PostgreSQL during ELevate

### PR DESCRIPTION
**Pull request by sloanebernstein**
Case RE-102, RE-637: Attempt to run an upgrade of the data directory for the system instance of PostgreSQL.

Changelog: Attempt to upgrade PostgreSQL during ELevate.

By submitting pull requests to this repo, I agree to the Contributor License Agreement which can be found at: https://github.com/cpanel/elevate/blob/main/docs/cPanel-CLA.pdf

### CODE REVIEW

**Review by cPholloway — APPROVED** [review]
Appoved pending any string changes from docs

File: lib/Elevate/Components/PostgreSQL.pm [hunk 1]
```diff
@@ -0,0 +1,394 @@
...
+    return if $self->_gave_up_on_postgresql;
+
+    INFO("Configuring PostgreSQL to work with cPanel's installation of phpPgAdmin.");
+    $self->service->start();    # less noisy when it's online, but still works
+
+    my ( $success, $msg ) = Whostmgr::Postgres::update_config();
+    if ( !$success ) {
+        ERROR("The system failed to update the PostgreSQL configuration: $msg");
+        Elevate::Notify::add_final_notification( <<~EOS );
+        ELevate was unable to configure the upgraded system PostgreSQL instance to work
```
Comment (WriterAmanda) [inline lib/Elevate/Components/PostgreSQL.pm:329]:
```suggestion
        ELevate could not configure the upgraded system PostgreSQL instance to work
```

File: lib/Elevate/Components/CCS.pm [hunk 2]
```diff
@@ -458,4 +413,64 @@ sub _import_data_for_single_user ( $self, $user ) {
     return;
 }
 
+=head1 move_pgsql_directory
+
+Due to the fact that CCS re-uses scripts originally intended for use with the
```
Comment (WriterAmanda) [inline lib/Elevate/Components/CCS.pm:418]:
```suggestion
Because CCS re-uses scripts originally intended for use with the
```

**Review by WriterAmanda — COMMENTED** [review]
I left 2 recommended wording changes. All else looks good.

**Review by cPholloway — APPROVED** [review]

**Review by annes449 — APPROVED** [review]
QA is all good.

**Pull request merged**